builtin/logical/transit/path_cache_config.go

```diff
@@ -0,0 +1,106 @@
+package transit
+
+import (
+	"context"
+	"errors"
+
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func (b *backend) pathCacheConfig() *framework.Path {
+	return &framework.Path{
+		Pattern: "cache-config",
+		Fields: map[string]*framework.FieldSchema{
+			"size": &framework.FieldSchema{
+				Type:        framework.TypeInt,
+				Required:    false,
+				Default:     0,
+				Description: `Size of cache, use 0 for an unlimited cache size, defaults to 0`,
+			},
+		},
+
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.ReadOperation: &framework.PathOperation{
+				Callback: b.pathCacheConfigRead,
+				Summary:  "Returns the size of the active cache",
+			},
+
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathCacheConfigWrite,
+				Summary:  "Configures a new cache of the specified size",
+			},
+
+			logical.CreateOperation: &framework.PathOperation{
+				Callback: b.pathCacheConfigWrite,
+				Summary:  "Configures a new cache of the specified size",
+			},
+		},
+
+		HelpSynopsis:    pathCacheConfigHelpSyn,
+		HelpDescription: pathCacheConfigHelpDesc,
+	}
+}
+
+func (b *backend) pathCacheConfigWrite(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	// get target size
+	cacheSize := d.Get("size").(int)
+	if cacheSize < 0 {
+		return nil, errors.New("size must be greater or equal to 0")
```

Comment:
```suggestion
		return logical.ErrorResponse("size must be greater or equal to 0"), logical.ErrInvalidRequest
```

Will cause it to return a 400 instead of a 500